### ISSUE TITLE
example mesh2d_manual crash

### ISSUE THREAD

**Issue by omarbassam88** [bevyengine/bevy#4036]
## Bevy version

0.6.0

## Operating system & version

Arch Linux with NVIDIA Geforce GTX 1060 6GB

## What you did

I ran the command:
```sh
cargo run --example mesh2d_manual
```

## What you expected to happen

A window should open with a 2D shape in it.

## What actually happened

and immediately crashed with the following error

```sh
thread 'TaskPool (11)' panicked at 'range end index 192 out of range for slice o
f length 176', crates/bevy_render/src/mesh/mesh/mod.rs:239:17
```

**Comment by alice-i-cecile** [comment]
Okay, that's working for me locally. So, some sort of driver or backend bug :/

When I run this example, I get the following line.

> INFO bevy_render::renderer: AdapterInfo { name: "NVIDIA GeForce GTX 970", vendor: 4318, device: 5058, device_type: DiscreteGpu, backend: Vulkan }

What's your version of that?

**Comment by mockersf** [comment]
are you sure you're on 0.6.0 and not on latest git version?

I can reproduce with main, it seems it's a regression introduced by e369a8ad5138af28a7e760fac3f07b278c27ebb4 (#3959)

**Comment by omarbassam88** [comment]
> Okay, that's working for me locally. So, some sort of driver or backend bug :/
> 
> When I run this example, I get the following line.
> 
> > INFO bevy_render::renderer: AdapterInfo { name: "NVIDIA GeForce GTX 970", vendor: 4318, device: 5058, device_type: DiscreteGpu, backend: Vulkan }
> 
> What's your version of that?

This is the version of the same line for me.

2022-02-25T16:35:51.237754Z  INFO bevy_render::renderer: AdapterInfo { name: "NVIDIA GeForce GTX 1060 6GB", vendor: 4318, device: 7171, device_type: DiscreteGpu, backend: Vulkan }

**Comment by omarbassam88** [comment]
> are you sure you're on 0.6.0 and not on latest git version?
> 
> I can reproduce with main, it seems it's a regression introduced by [e369a8a](https://github.com/bevyengine/bevy/commit/e369a8ad5138af28a7e760fac3f07b278c27ebb4) (#3959)

Yes, I guess you are right I am on main branch and I just checked the commit you mentioned regarding the change of handling of vertex attribute.
The example custom_vertex_attribute works fine for me. So I guess it's the `mesh2d_manual` example that needs to be updated maybe? if that's the case i can try to work on it if needed.

**Comment by mockersf** [comment]
there is a fix in #4037 you could try 🙂

**Comment by omarbassam88** [comment]
> there is a fix in #4037 you could try slightly_smiling_face

I did check check this pull request and I can confirm that it does fix the problem.

**Comment by KDecay** [comment]
I just encountered a problem while creating a mesh and tried the `mesh2d_manual` to see what I was missing and it also didn't work for me on `main`. I can also confirm that the fix in #4037 works.
